src/extension/importer/core/src/main/java/org/geoserver/importer/transform/AttributeComputeTransform.java

```diff
@@ -0,0 +1,87 @@
+/* (c) 2018 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.importer.transform;
+
+import org.geoserver.importer.ImportTask;
+import org.geotools.data.DataStore;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.filter.text.cql2.CQLException;
+import org.geotools.filter.text.ecql.ECQL;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.filter.expression.Expression;
+
+/**
+ * Transform creating a new attribute based on the existing ones
+ */
+public class AttributeComputeTransform extends AbstractTransform implements InlineVectorTransform {
+    
+    private static final long serialVersionUID = 1L;
+
+    /** field to remap */
+    protected String field;
+
+    /** type to remap to */
+    protected Class type;
+    
+    /** the expression to apply (stored as a string as CQL does not always round trip properly */
+    protected String cql;
+    protected Expression expression;
+    
+    public AttributeComputeTransform(String field, Class type, String cql) throws CQLException {
+        this.field = field;
+        this.type = type;
+        this.cql = cql;
+        this.expression = ECQL.toExpression(cql);
+    }
+    
+    protected AttributeComputeTransform() {
+        
+    }
+
+    public String getField() {
+        return field;
+    }
+
+    public void setField(String field) {
+        this.field = field;
+    }
+
+    public Class getType() {
+        return type;
+    }
+
+    public void setType(Class type) {
+        this.type = type;
+    }
+
+    public String getCql() {
+        return cql;
+    }
+
+    public void setCql(String cql) throws CQLException {
+        this.cql = cql;
+        this.expression = ECQL.toExpression(cql);
+    }
+
+    public SimpleFeatureType apply(ImportTask task, DataStore dataStore,
+                                   SimpleFeatureType featureType) throws Exception {
+        // remap the type
+        SimpleFeatureTypeBuilder builder = new SimpleFeatureTypeBuilder();
```

Comment:
I would check if the attribute is indeed a new attribute and don't already exists. If the attribute already exists I guess the result will depend on the underling data source. I made a quick test and with a relational database it will fail complaining about  repeated column:

```
12 Jan 11:31:16 ERROR [geoserver.importer] - Error occured during import
java.io.IOException: Error occurred creating table
	at org.geotools.jdbc.JDBCDataStore.createSchema(JDBCDataStore.java:822)
	at org.geotools.jdbc.JDBCDataStore.createSchema(JDBCDataStore.java:148)
	at org.geoserver.importer.Importer.loadIntoDataStore(Importer.java:1262)
	at org.geoserver.importer.Importer.doIndirectImport(Importer.java:1033)
	at org.geoserver.importer.Importer.run(Importer.java:859)
	at org.geoserver.importer.Importer.run(Importer.java:809)
	at org.geoserver.importer.Importer.run(Importer.java:783)
	at org.geoserver.importer.Importer.run(Importer.java:779)
	at org.geoserver.importer.rest.ImporterIntegrationTest.testAttributeCompute(ImporterIntegrationTest.java:526)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: org.h2.jdbc.JdbcSQLException: Duplicate column name CAT_DESC; SQL statement:
CREATE TABLE "archsites" ( "fid" int AUTO_INCREMENT(1) PRIMARY KEY, "the_geom" BYTEA COMMENT 'POINT', "ID" BIGINT, "CAT_ID" BIGINT, "CAT_DESC" VARCHAR(255), "CAT_DESC" VARCHAR(255) )  [42121-119]
	at org.h2.message.Message.getSQLException(Message.java:106)
	at org.h2.message.Message.getSQLException(Message.java:117)
	at org.h2.message.Message.getSQLException(Message.java:76)
	at org.h2.table.Table.setColumns(Table.java:335)
	at org.h2.table.TableData.<init>(TableData.java:82)
	at org.h2.schema.Schema.createTable(Schema.java:492)
	at org.h2.command.ddl.CreateTable.update(CreateTable.java:139)
	at org.h2.command.CommandContainer.update(CommandContainer.java:72)
	at org.h2.command.Command.executeUpdate(Command.java:208)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:162)
	at org.apache.commons.dbcp.DelegatingStatement.execute(DelegatingStatement.java:264)
	at org.apache.commons.dbcp.DelegatingStatement.execute(DelegatingStatement.java:264)
	at org.geotools.jdbc.JDBCDataStore.createSchema(JDBCDataStore.java:814)
	... 35 more
```